File: An 2/Sem 1/MAP/Lab10/C#/Lab2/Model/UndergraduateStudent.cs
using System;

namespace Lab2
{
	namespace Lab2_Model {
		[Serializable()]
		class UndergraduateStudent : Student, Comparable<Student>, Readable
		{
            public int grade2 { get; set; }

			public UndergraduateStudent() : base() {
			}

			public UndergraduateStudent(int id, String name, int grade, int grade2) : base(id, name, grade) {
				this.grade2 = grade2;
			}

			public override float average() {
				return (this.grade + this.grade2) / 2;
			}

			bool Comparable<Student>.isGreaterThan(Student student) {
				return (this.average() > student.average());
			}


			public override String ToString() {
				return String.Format("{0}|{1}|{2}|{3}|{4}|", this.GetType(), this.id, this.name, this.grade, this.grade2);
		    }

			void Readable.readAttributesFromString(String inputString) {
				string[] tokens = inputString.Split ('|');
				this.id = Convert.ToInt16 (tokens [1]);
				this.name = tokens [2];
				this.grade = Convert.ToInt16 (tokens [3]);
				this.grade2 = Convert.ToInt16 (tokens [4]);
			}
		}
	}
}

File: An 2/Sem 2/AOP/Lab 1/src/com/company/Model/UndergraduateStudent.java
package com.company.Model;

import java.util.logging.Logger;

public class UndergraduateStudent extends Student {
    public int grade2;
    private static Logger logger = Logger.getLogger("Students");


    public UndergraduateStudent() {
        super();
    }

    public UndergraduateStudent(int id, String name, int grade, int grade2) {
        super(id, name, grade);
        logger.info("[Entering:]UndergraduateStudent.init");
        this.grade2 = grade2;
    }

    @Override
    public float average() {
        logger.info("[Entering:]UndergraduateStudent.average");
        return (this.grade + this.grade2) / 2;
    }

    @Override
    public boolean isGreaterThan(Student student) {
        logger.info("[Entering:]UndergraduateStudent.isGreaterThan");
        return (this.average() > student.average());
    }

    @Override
    public String toString() {
        logger.info("[Entering:]UndergraduateStudent.toString");
        return String.format("%s|%d|%s|%d|%d|", this.getClass().toString(), this.id, this.name, this.grade, this.grade2);
    }

    @Override
    public void readAttributesFromString(String string) {
        logger.info("[Entering:]UndergraduateStudent.readAttributesFromString");
        String[] tokens = string.split("[|]+");
        this.id = Integer.parseInt(tokens[1]);
        this.name = tokens[2];
        this.grade = Integer.parseInt(tokens[3]);
        this.grade2 = Integer.parseInt(tokens[4]);
    }
}
